### DISCUSSION TITLE
(not urgent) I pooched my installation when updating

### DISCUSSION THREAD

**Discussion by HowardCrane** [AzuraCast/AzuraCast#4990] [General]
Hi folks,
I could just wipe my Ubuntu 18.04.6 and reinstall Azuracast because it's only one station and I'm a fan of starting over, but I'd like to solve this issue for the sake of understanding my software, and I don't want to monkey around with my API again. 
I installed and updated using Docker and skipped from 0.9.8.1 to today's version - rookie mistake. The site is unreachable, I tried restarting nginx but the service isn't found (!?). Might have sometime to do with that time I tried to get Apache to play with nginx.

Should I just reinstall?

I got these wanrings:

ERROR: Invalid interpolation format for "installer" option in service "services": "ghcr.io/azuracast/web:${AZURACAST_VERSION:-latest}"
ERROR: Invalid interpolation format for "installer" option in service "services": "ghcr.io/azuracast/web:${AZURACAST_VERSION:-latest}"
ERROR: Invalid interpolation format for "nginx_proxy" option in service "services": "ghcr.io/azuracast/nginx_proxy:${AZURACAST_VERSION:-latest}"

My complete update went like this:

Are you ready to continue with the update? [Y/n] y
services:
  influxdb:
    image: azuracast/azuracast_influxdb:latest
    logging:
      options:
        max-file: '5'
        max-size: 1m
    restart: always
    volumes:
    - influx_data:/var/lib/influxdb:rw
  mariadb:
    environment:
      APPLICATION_ENV: production
      AUTO_ASSIGN_PORT_MAX: '8499'
      AUTO_ASSIGN_PORT_MIN: '8000'
      AZURACAST_DC_REVISION: 7
      AZURACAST_SFTP_PORT: '2022'
      COMPOSER_PLUGIN_MODE: "false"
      LANG: en_US.UTF-8
      MYSQL_DATABASE: azuracast
      MYSQL_HOST: mariadb
      MYSQL_PASSWORD: azur4c457
      MYSQL_PORT: '3306'
      MYSQL_RANDOM_ROOT_PASSWORD: "yes"
      MYSQL_USER: azuracast
      PREFER_RELEASE_BUILDS: "false"
    image: azuracast/azuracast_db:latest
    logging:
      options:
        max-file: '5'
        max-size: 1m
    restart: always
    volumes:
    - db_data:/var/lib/mysql:rw
  nginx_proxy:
    depends_on:
      web:
        condition: service_started
    environment:
      DEFAULT_HOST: azuracast.local
    image: azuracast/azuracast_nginx_proxy:latest
    ports:
    - 80:80/tcp
    - 443:443/tcp
    restart: always
    volumes:
    - letsencrypt:/etc/nginx/certs:rw
    - nginx_proxy_vhosts:/etc/nginx/vhost.d:rw
    - /var/run/docker.sock:/tmp/docker.sock:ro
    - letsencrypt_html:/usr/share/nginx/html:rw
  nginx_proxy_letsencrypt:
    image: jrcs/letsencrypt-nginx-proxy-companion:latest
    restart: always
    volumes:
    - /var/run/docker.sock:/var/run/docker.sock:ro
    volumes_from:
    - service:nginx_proxy:rw
  redis:
    image: azuracast/azuracast_redis:latest
    logging:
      options:
        max-file: '5'
        max-size: 1m
    restart: always
    sysctls:
      net.core.somaxconn: '1024'
    volumes:
    - redis_data:/data:rw
  stations:
    container_name: azuracast_stations
    image: azuracast/azuracast_radio:latest
    init: true
    logging:
      options:
        max-file: '5'
        max-size: 1m
    ports:
    - 8000:8000/tcp
    - 8005:8005/tcp
    - 8006:8006/tcp
    - 8010:8010/tcp
    - 8015:8015/tcp
    - 8016:8016/tcp
    - 8020:8020/tcp
    - 8025:8025/tcp
    - 8026:8026/tcp
    - 8030:8030/tcp
    - 8035:8035/tcp
    - 8036:8036/tcp
    - 8040:8040/tcp
    - 8045:8045/tcp
    - 8046:8046/tcp
    - 8050:8050/tcp
    - 8055:8055/tcp
    - 8056:8056/tcp
    - 8060:8060/tcp
    - 8065:8065/tcp
    - 8066:8066/tcp
    - 8070:8070/tcp
    - 8075:8075/tcp
    - 8076:8076/tcp
    - 8090:8090/tcp
    - 8095:8095/tcp
    - 8096:8096/tcp
    - 8100:8100/tcp
    - 8105:8105/tcp
    - 8106:8106/tcp
    - 8110:8110/tcp
    - 8115:8115/tcp
    - 8116:8116/tcp
    - 8120:8120/tcp
    - 8125:8125/tcp
    - 8126:8126/tcp
    - 8130:8130/tcp
    - 8135:8135/tcp
    - 8136:8136/tcp
    - 8140:8140/tcp
    - 8145:8145/tcp
    - 8146:8146/tcp
    - 8150:8150/tcp
    - 8155:8155/tcp
    - 8156:8156/tcp
    - 8160:8160/tcp
    - 8165:8165/tcp
    - 8166:8166/tcp
    - 8170:8170/tcp
    - 8175:8175/tcp
    - 8176:8176/tcp
    - 8180:8180/tcp
    - 8185:8185/tcp
    - 8186:8186/tcp
    - 8190:8190/tcp
    - 8195:8195/tcp
    - 8196:8196/tcp
    - 8200:8200/tcp
    - 8205:8205/tcp
    - 8206:8206/tcp
    - 8210:8210/tcp
    - 8215:8215/tcp
    - 8216:8216/tcp
    - 8220:8220/tcp
    - 8225:8225/tcp
    - 8226:8226/tcp
    - 8230:8230/tcp
    - 8235:8235/tcp
    - 8236:8236/tcp
    - 8240:8240/tcp
    - 8245:8245/tcp
    - 8246:8246/tcp
    - 8250:8250/tcp
    - 8255:8255/tcp
    - 8256:8256/tcp
    - 8260:8260/tcp
    - 8265:8265/tcp
    - 8266:8266/tcp
    - 8270:8270/tcp
    - 8275:8275/tcp
    - 8276:8276/tcp
    - 8280:8280/tcp
    - 8285:8285/tcp
    - 8286:8286/tcp
    - 8290:8290/tcp
    - 8295:8295/tcp
    - 8296:8296/tcp
    - 8300:8300/tcp
    - 8305:8305/tcp
    - 8306:8306/tcp
    - 8310:8310/tcp
    - 8315:8315/tcp
    - 8316:8316/tcp
    - 8320:8320/tcp
    - 8325:8325/tcp
    - 8326:8326/tcp
    - 8330:8330/tcp
    - 8335:8335/tcp
    - 8336:8336/tcp
    - 8340:8340/tcp
    - 8345:8345/tcp
    - 8346:8346/tcp
    - 8350:8350/tcp
    - 8355:8355/tcp
    - 8356:8356/tcp
    - 8360:8360/tcp
    - 8365:8365/tcp
    - 8366:8366/tcp
    - 8370:8370/tcp
    - 8375:8375/tcp
    - 8376:8376/tcp
    - 8380:8380/tcp
    - 8385:8385/tcp
    - 8386:8386/tcp
    - 8390:8390/tcp
    - 8395:8395/tcp
    - 8396:8396/tcp
    - 8400:8400/tcp
    - 8405:8405/tcp
    - 8406:8406/tcp
    - 8410:8410/tcp
    - 8415:8415/tcp
    - 8416:8416/tcp
    - 8420:8420/tcp
    - 8425:8425/tcp
    - 8426:8426/tcp
    - 8430:8430/tcp
    - 8435:8435/tcp
    - 8436:8436/tcp
    - 8440:8440/tcp
    - 8445:8445/tcp
    - 8446:8446/tcp
    - 8450:8450/tcp
    - 8455:8455/tcp
    - 8456:8456/tcp
    - 8460:8460/tcp
    - 8465:8465/tcp
    - 8466:8466/tcp
    - 8470:8470/tcp
    - 8475:8475/tcp
    - 8476:8476/tcp
    - 8480:8480/tcp
    - 8485:8485/tcp
    - 8486:8486/tcp
    - 8490:8490/tcp
    - 8495:8495/tcp
    - 8496:8496/tcp
    restart: always
    ulimits:
      nofile:
        hard: 65536
        soft: 65536
    volumes:
    - letsencrypt:/etc/nginx/ssl:ro
    - shoutcast2_install:/var/azuracast/servers/shoutcast2:rw
    - station_data:/var/azuracast/stations:rw
    - tmp_data:/var/azuracast/www_tmp:rw
  web:
    container_name: azuracast_web
    depends_on:
      influxdb:
        condition: service_started
      mariadb:
        condition: service_started
      redis:
        condition: service_started
      stations:
        condition: service_started
    environment:
      APPLICATION_ENV: production
      AUTO_ASSIGN_PORT_MAX: '8499'
      AUTO_ASSIGN_PORT_MIN: '8000'
      AZURACAST_DC_REVISION: 7
      AZURACAST_SFTP_PORT: '2022'
      COMPOSER_PLUGIN_MODE: "false"
      LANG: en_US.UTF-8
      MYSQL_DATABASE: azuracast
      MYSQL_HOST: mariadb
      MYSQL_PASSWORD: azur4c457
      MYSQL_PORT: '3306'
      MYSQL_RANDOM_ROOT_PASSWORD: "yes"
      MYSQL_USER: azuracast
      PREFER_RELEASE_BUILDS: "false"
      VIRTUAL_HOST: evmedia.net
    image: azuracast/azuracast_web_v2:latest
    logging:
      options:
        max-file: '5'
        max-size: 1m
    ports:
    - 10080:80/tcp
    - 10443:443/tcp
    - 2022:2022/tcp
    restart: always
    ulimits:
      nofile:
        hard: 65536
        soft: 65536
    volumes:
    - letsencrypt:/etc/letsencrypt:rw
    - backups:/var/azuracast/backups:rw
    - geolite_install:/var/azuracast/geoip:rw
    - shoutcast2_install:/var/azuracast/servers/shoutcast2:rw
    - sftpgo_data:/var/azuracast/sftpgo/persist:rw
    - station_data:/var/azuracast/stations:rw
    - www_data:/var/azuracast/www:rw
    - tmp_data:/var/azuracast/www_tmp:rw
version: '2.2'
volumes:
  backups: {}
  db_data: {}
  geolite_install: {}
  influx_data: {}
  letsencrypt: {}
  letsencrypt_html: {}
  nginx_proxy_vhosts: {}
  redis_data: {}
  sftpgo_data: {}
  shoutcast2_install: {}
  station_data: {}
  tmp_data: {}
  www_data: {}

ERROR: Invalid interpolation format for "installer" option in service "services": "ghcr.io/azuracast/web:${AZURACAST_VERSION:-latest}"
ERROR: Invalid interpolation format for "installer" option in service "services": "ghcr.io/azuracast/web:${AZURACAST_VERSION:-latest}"
ERROR: Invalid interpolation format for "nginx_proxy" option in service "services": "ghcr.io/azuracast/nginx_proxy:${AZURACAST_VERSION:-latest}"
Stopping azuracast_nginx_proxy_letsencrypt_1 ... done
Stopping azuracast_nginx_proxy_1             ... done
Stopping azuracast_web                       ... done
Stopping azuracast_stations                  ... done
Stopping azuracast_redis_1                   ... done
Stopping azuracast_mariadb_1                 ... done
Stopping azuracast_influxdb_1                ... done
Removing azuracast_nginx_proxy_letsencrypt_1 ... done
Removing azuracast_nginx_proxy_1             ... done
Removing azuracast_web                       ... done
Removing azuracast_stations                  ... done
Removing azuracast_redis_1                   ... done
Removing azuracast_mariadb_1                 ... done
Removing azuracast_influxdb_1                ... done
Removing network azuracast_default
Error: No such volume: azuracast_www_vendor
azuracast_tmp_data
azuracast_redis_data
ERROR: Version mismatch: file ./docker-compose.yml specifies version 1 but extension file ./docker-compose.override.yml uses version 2.2
ERROR: Version mismatch: file ./docker-compose.yml specifies version 1 but extension file ./docker-compose.override.yml uses version 2.2
Clean up all stopped Docker containers and images to save space? [Y/n] n
Update complete!

**Comment by BusterNeece** [comment]
@HowardCrane Quite a lot has changed in the last few versions, but you should still be able to jump directly from an 0.9 version to a current version just by making these changes:

- Make sure to run `./docker.sh update-self`
- Run `./docker.sh install-docker-compose` to make sure the latest Docker Compose is present
- Update your `docker-compose.override.yml` file to remove the top line (the "version" string)

After that, if you re-run the updater it should run fine.

**Comment by HowardCrane** [comment]
@SlvrEagle23 Thanks! The update went smoothly. I'm just left with a 503 error on the browser. Preliminary web search suggests this is LetsEncrypt not playing ball. Should I post in "Issues"? If so, which log files should I include? Nginx and LetsEncrypt mystifies me. Would me using the IP of my server as an address conflict? I don't kow if it's treated as a domain name.
Thanks for the help!

**Reply by BusterNeece** [reply]
@HowardCrane You may want to run `./docker.sh letsencrypt-create` just to set up LetsEncrypt again; the format we use may have changed in the intervening versions.

**Reply by HowardCrane** [reply]
@SlvrEagle23 Yes, that needed to be set up manually/separately. You'd never see what happened next - Azuracast and docker vanished from my server lol the directory /etc/azuracast/ existed right up until I ran the LetsEncrypt installer, so it's definitely related.
I'll just wipe the server and start over later. Thanks for the help. Hope I supplied meaningful data on the effects of skipping big version changes.

**Reply by BusterNeece** [reply]
@HowardCrane It should be `/var/azuracast` actually...that's the directory it typically lives in. I'm not sure why it would just disappear (it really shouldn't do that).

**Reply by HowardCrane** [reply]
@SlvrEagle23 Yes I'm sorry, I mistyped in that message above. There's no sign of anything left on my Ubuntu. It's bizarre
